### PR TITLE
Added reject transactions modal to be present in the footer of the approve screen

### LINKED ISSUE #16769: [Bug]: Users cannot reject and remove all unapproved `approve` transactions
### Describe the bug

`Reject X Transactions` button is no longer present in the footer of the `approve` screen

Before: Batch reject button present

https://user-images.githubusercontent.com/53189696/205168516-fbb16f2d-9ca0-49ac-8800-a190e7a5262e.mov

Now: Missing batch reject button

https://user-images.githubusercontent.com/53189696/205168556-cfb8850a-9a63-42f7-a23a-ad02348ed7ff.mov

### Steps to reproduce

1. Navigate to the [test dapp](https://metamask.github.io/test-dapp)
2. Connect your wallet
3. Deploy the erc20 token
4. Click `Approve Token` multiple times
5. Navigate back to the wallet

### Error messages or log output

_No response_

### Version

10.23.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

Added reject transactions modal to be present in the footer of the approve screen. Batch reject button is present now in the token allowance screen.

* Fixes #16769 

#### test-dapp

https://user-images.githubusercontent.com/92527393/206162292-1ac910eb-090d-48b5-a3b8-25ab5e535b35.mov

#### etherscan

https://user-images.githubusercontent.com/92527393/206162516-0803a2fa-1800-4b11-a2c4-a265adecc166.mov

### test-dapp

1. Navigate to the [test dapp](https://metamask.github.io/test-dapp)
2. Connect your wallet
3. Deploy the `ERC20` token
4. Click `Approve Token` multiple times
5. Navigate back to the wallet

### etherscan

1. Go to https://etherscan.io/token/0x6b175474e89094c44da98b954eedeac495271d0f#writeContract
2. Connect your wallet
3. Under `approve` input any address and a number
4. Click on `write` multiple times
5. Navigate back to the wallet

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

File: ui/pages/token-allowance/token-allowance.js [hunk 1]
```diff
@@ -413,7 +434,13 @@ export default function TokenAllowance({
         onCancel={() => handleReject()}
         onSubmit={() => (isFirstPage ? handleNextClick() : handleApprove())}
         disabled={disableNextButton || disableApproveButton}
-      />
+      >
+        {unapprovedTxCount > 1 && (
+          <a onClick={() => handleCancelAll()}>
```
Comment (darkwing) [inline ui/pages/token-allowance/token-allowance.js:439]: To make this accessible, this element needs a `href` attribute. So maybe something like:

```
<a href="" onClick={(e) => { e.preventDefault(); handleCancelAll() }}>
```

Better yet would be to use a `<Button>` component.

**Review by darkwing — CHANGES_REQUESTED** [review]

**Review by darkwing — APPROVED** [review]
Nicely done @VSaric !

**Comment by jpuri** [comment]
Great work 👍 

Will be nice to have some test also added.

**Comment by VSaric** [comment]
Hey @jpuri, I agree on adding some test for this, but we don't have any tests for `token-allowance.js`. Is it okay for you that we create new PR for adding tests for `token-allowance.js` and move forward with this PR and ship it? 

cc: @bschorchit

**Comment by danjm** [comment]
> Is it okay for you that we create new ticket for adding tests for token-allowance.js and move forward with this PR and ship it?

I think this would be okay in this case. This just fixes a part of the component and is sort of a blocker for next release. But we should prioritize the tests very soon.

**Comment by bschorchit** [comment]
I've created the new ticket for adding tests to token allowance here https://github.com/MetaMask/metamask-extension/issues/16910 and added to PS team's board.

**Review by danjm — APPROVED** [review]

**Pull request merged**